### PR TITLE
Add tree based machine learning model

### PR DESCRIPTION
**Pull request by ninhgiang225**
# Description

Created additional feature: tree-based machine learning by adding pages and algorithm for tree-based machine learning model

## Type of Change

- make tree_based_ml directory
- add xgboost_model.py in tree_based_ml
- add gradient_boosting.py in tree_based_ml
- add catboost_model.py in tree_based_ml
- add adaboost.py in tree_based_ml
- add lightbgm_model.py in tree_based_ml
- add radom_forest.py in tree_based_ml
- add new author: Gina Nguyen

---

> [!NOTE]
> **Medium Risk**
> Adds a new Streamlit page with multiple interactive training flows and introduces heavy native ML dependencies (`xgboost`, `lightgbm`, `catboost`), increasing runtime/packaging risk. Also commits generated `catboost_info/*` training artifacts which can bloat the repo and create noisy diffs.
> 
> **Overview**
> **Introduces a new Tree-Based ML area in the Streamlit app.** Adds `pages/7_Tree_based_Machine_Learning.py` with a sidebar selector and overview, routing into new model modules for Random Forest, Gradient Boosting, AdaBoost, XGBoost, LightGBM, and CatBoost.
> 
> Each model module (`tree_based_ml/*`) includes interactive theory visualizations plus a “train on your uploaded CSV/Excel” workflow with basic metrics and plots.
> 
> **Dependencies and metadata updated.** `pyproject.toml`/`uv.lock` add `xgboost`, `lightgbm`, and `catboost` (plus transitive deps), and the footer/README update author attribution; `catboost_info/*` training output files are added to the repo.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2b3955c93abf37b6bc3023d4eba0e8da167f75b4. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by MarkPhamm** [comment]
@ninhgiang225 does these file belong to .gitignore?
image

**Comment by MarkPhamm** [comment]
also pls fix lint

**Comment by MarkPhamm** [comment]
@cursor review

**Review by ninhgiang225 — COMMENTED** [review]
Only the styles and spacing of the code being changed, such that it order the imported package alphabetically and delete the redundant spacing

**Review by MarkPhamm — APPROVED** [review]

**Comment by MarkPhamm** [comment]
@cursor fix lint: 
```
Added /opt/hostedtoolcache/ruff/0.15.6/x86_64 to the path
Set RUFF_OUTPUT_FORMAT to github
Successfully installed ruff version 0.15.6
/opt/hostedtoolcache/ruff/0.15.6/x86_64/ruff check /home/runner/work/Stats-Calculator/Stats-Calculator
Error: tree_based_ml/adaboost.py:60:28: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/adaboost.py:104:28: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/adaboost.py:147:55: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/adaboost.py:150:37: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/adaboost.py:161:46: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/adaboost.py:162:37: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/catboost_model.py:69:41: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/catboost_model.py:73:28: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/catboost_model.py:105:40: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/catboost_model.py:107:35: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/catboost_model.py:109:28: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/catboost_model.py:171:46: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/catboost_model.py:172:37: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/catboost_model.py:179:33: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:58:45: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:60:35: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:62:28: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:89:49: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:91:39: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:93:32: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:139:55: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:141:38: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:153:46: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:154:37: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/gradient_boosting.py:160:33: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/lightgbm_model.py:68:41: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/lightgbm_model.py:72:28: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/lightgbm_model.py:108:36: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/lightgbm_model.py:110:35: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/lightgbm_model.py:112:28: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/lightgbm_model.py:166:46: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/lightgbm_model.py:167:37: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/lightgbm_model.py:174:33: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/random_forest.py:60:46: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/random_forest.py:62:25: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/random_forest.py:146:46: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/random_forest.py:147:37: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/random_forest.py:153:33: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:72:36: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:74:35: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:76:28: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:110:54: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:112:25: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:114:28: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:168:46: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:169:35: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:170:38: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:184:46: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:185:37: E702 Multiple statements on one line (semicolon)
Error: tree_based_ml/xgboost_model.py:191:33: E702 Multiple statements on one line (semicolon)
Error: The process '/opt/hostedtoolcache/ruff/0.15.6/x86_64/ruff' failed with exit code 1
```
